src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -327,6 +342,7 @@ internal NodeAutoCompleteBarViewModel(DynamoViewModel dynamoViewModel) : base(dy
             // Off load some time consuming operation here
             DefaultResults = dynamoViewModel.DefaultAutocompleteCandidates.Values;
             ServiceVersion = string.Empty;
+            dynamoViewModel.CurrentSpaceViewModel.Model.NodeRemoved += NodeViewModel_Removed;
```

Comment:
We keep a single instance of the BarViewModel so after first close of the bar we won't be subscribed anymore to this event. Since we only want to close the bar when the query node is deleted I would say to move this to the NodeAutoCompleteBarView and handle it in the same way as the other workspace events.

Reply:
Good catch, fixed and changed the event management to a clearer process